docs/dev/table/sqlClient.md

```diff
@@ -0,0 +1,538 @@
+     -l,--library <JAR directory>          A JAR file directory with which every
+                                           new session is initialized. The files
+                                           might contain user-defined classes
+                                           needed for the execution of
+                                           statements such as functions, table
+                                           sources, or sinks. Can be used
+                                           multiple times.
+     -s,--session <session identifier>     The identifier for a session.
+                                           'default' is the default identifier.
+{% endhighlight %}
+
+{% top %}
+
+### Environment Files
+
+A SQL query needs a configuration environment in which it is executed. The so-called *environment files* define available table sources and sinks, external catalogs, user-defined functions, and other properties required for execution and deployment.
+
+Every environment file is a regular [YAML file](http://http://yaml.org/) that looks similar to the following example. The file defines an environment with a table source `MyTableName` that reads from CSV file. Queries that are executed in this environment (among others) will have a parallelism of 1, an even-time characteristic, and will run in the `table` result mode.
+
+{% highlight yaml %}
+# Define table sources and sinks here.
+
+tables:
+  - name: MyTableName
+    type: source
+    schema:
+      - name: MyField1
+        type: INT
+      - name: MyField2
+        type: VARCHAR
+    connector:
+      type: filesystem
+      path: "/path/to/something.csv"
+    format:
+      type: csv
+      fields:
+        - name: MyField1
+          type: INT
+        - name: MyField2
+          type: VARCHAR
+      line-delimiter: "\n"
+      comment-prefix: "#"
+
+# Execution properties allow for changing the behavior of a table program.
+
+execution:
+  type: streaming
+  time-characteristic: event-time
+  parallelism: 1
+  max-parallelism: 16
+  min-idle-state-retention: 0
+  max-idle-state-retention: 0
+  result-mode: table
+
+# Deployment properties allow for describing the cluster to which table programs are submitted to.
+
+deployment:
+  response-timeout: 5000
+{% endhighlight %}
+
+Environment files can be created for general purposes (*defaults environment file* using `--defaults`) as well as on a per-session basis (*session environment file* using `--environment`). Every CLI session is initialized with the default properties followed by the session properties. Both default and session environment files can be passed when starting the CLI application. If no default environment file has been specified, the SQL Client searches for `./conf/sql-client-defaults.yaml` in Flink's configuration directory.
+
+Properties that have been set within a CLI session (e.g. using the `SET` command) have highest precedence:
+
```

Comment:
This can be a banner or sth more highlighted, with the title "Configuration Priorities" using either:
```<span class="label label-danger">Attention</span>```
or 
```
<div class="alert alert-info">
  <strong>Note:</strong> ....
</div>
```